modules/serverless-kafka-tls.adoc

```diff
@@ -0,0 +1,46 @@
+// Module is included in the following assemblies:
+//
+// serverless/serverless-kafka.adoc
+
+[id="serverless-configuring-tls-authentication-against-an-apache-kafka_{context}"]
+== Configuring TLS authentication
+
+.Prerequisites
+
+* A Kafka cluster CA certificate as a `.pem` file.
+* A Kafka cluster client certificate and key as `.pem` files.
+
+.Procedure
+
+. Create the certificate files as secrets in your chosen namespace:
++
+----
+$ kubectl create secret --namespace <namespace> generic <kafka-auth-secret> \
+  --from-file=ca.crt=caroot.pem \
+  --from-file=user.crt=certificate.pem \
+  --from-file=user.key=key.pem
+----
++
+[IMPORTANT]
+====
+Use the key names `ca.crt`, `user.crt`, and `user.key`. Do not change them.
+====
+
+. Reference your secret and the namespace of the secret in the `config-kafka` config map:
++
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: config-kafka
+  namespace: knative-eventing
+data:
+  bootstrapServers: <list_of_bootstrap_servers>
+  authSecretName: <kafka_auth_secret>
+  authSecretNamespace: <namespace>
+----
```

Comment:
Same here. We should advise editing `KnativeKafka` CR instead.

Reply:
@cardil How would this step look exactly?

Reply:
For TLS it should be:

```yaml
apiVersion: operator.serverless.openshift.io/v1alpha1
kind: KnativeKafka
metadata:
  namespace: knative-eventing
  name: knative-kafka
spec:
  channel:
    authSecretName: tls-user
    authSecretNamespace: kafka
    bootstrapServers: eventing-kafka-bootstrap.kafka.svc:9093
    enabled: true
  source:
    enabled: true
```

Note, I'm here using a TLS listener (port 9093) as tls auth is only available on TLS listener.

@aliok Could you confirm those settings makes sense?

Reply:
@cardil Here as well cc @aliok

Reply:
https://deploy-preview-29059--osdocs.netlify.app/openshift-enterprise/latest/serverless/serverless-kafka.html#serverless-configuring-tls-authentication-against-an-apache-kafka_serverless-kafka

Reply:
@cardil thanks for noticing that.

@rh-max yes, customers should use `KnativeKafka` CR. Things in that CR is propagated into `config-kafka` configmap.

If they change the configmap, the operator might revert their changes.

@cardil 's suggestion is correct.

More info on https://github.com/openshift-knative/serverless-operator/pull/699

```
authSecretName: <kafka-auth-secret-that-the-user-created-before>
authSecretNamespace: <kafka-auth-secret-namespace>
bootstrapServers: <bootstrap-server>
```

As Chris pointed out, user needs to make sure they're giving the right ports.

e.g., in following file, Kafka is setup to serve TLS on port 9093: https://github.com/strimzi/strimzi-kafka-operator/blob/main/examples/security/tls-auth/kafka.yaml

but it is possible to listen to another port and use SASL there.

Reply:
Thanks @aliok for the explanation!